03week/towersOfHanoi.js

```diff
@@ -19,24 +19,71 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+/* Code Plan 
+  // Tower of Hanoi Rules
+    // Goal is to move all disks from left most peg to the right most peg
+    // Only one disk may be moved at a time
+    // A move is moving one disk from the top of a stack and then placing in on an empty peg or another stack
+    // Disks cannot be placed on smaller disk than the one that is moving
+  // movePiece()
+    // move a disk from the top of one stack, and place it at the end of another stack. .pop() and .push()
+  // isLegal()
+    // function should make sure only one disk is moving, and the destination peg is empty or the top disk of destination stack is smaller than the moved disk stack.length() and the
+  // checkForWin()
+    // checks to see if the middle or right stack is in descending order from largest to smallest [4,3,2,1]
+    // just checking if array == to array will not work. have to manually check the values of the array using a loop
+  // resetGame()
+    // restarts that game and clears all stacks except for farthest left stack, and makes it in descending order largest to smallesr [4,3,2,1]
+  //towersOfHanoi
+    // parent function that calls other critical functions of the game
+    // check if the move is a legal move isLegal()
+    // if it is a legal move than move the disk
+    // check for a win after the move
+    // if the player wins the game, reset the game, set stacks object to values before game started
+*/
+function resetGame(){
+  stacks.a = [4,3,2,1];
+  stacks.b = [];
+  stacks.c = [];
+}
+function movePiece(from, to) {
+  stacks[to].push(stacks[from].pop());
 }
 
-function isLegal() {
-  // Your code here
-
+function isLegal(startStack, endStack) {
+  const possibleStacks = ['a','b','c'];
+  if( possibleStacks.includes(startStack, endStack ) ){
+    const fromValue = stacks[startStack][stacks[startStack].length -1]
+    const toValue =  stacks[endStack][stacks[endStack].length -1];
+    return ( toValue == undefined || fromValue < toValue );
+  } else {
+    return false
+  }
 }
 
 function checkForWin() {
-  // Your code here
-
-}
+  const descStack = [4,3,2,1];
+  if(stacks['b'].length !== descStack.length && stacks['c'].length !== descStack.length){
+    return false;
+    for(let i = descStack.length; i--;) {
+      if(stacks['b'][i] !== descStack[i] || stacks['c'][i] !== descStack[i]){
+        return false;
+      }
+    }
+  }
+  console.log('You Win, resetting the game.');
+  resetGame();
```

Comment:
Your check for win function should not be calling the restGame function, it should only be checking for a win